File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        cur = prices[0]
        total = 0

        for price in prices:
            if cur < price:
                total += price - cur
            
            cur = price
        
        return total
        

File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int curr = prices[0];
        int total = 0;

        for(int price : prices){
            if(curr < price){
                total += price - curr;
            }
            curr = price;
        }

        return total;
    }
};